src/main/webapp/app/quiz/re-evaluate/multiple-choice-question/re-evaluate-multiple-choice-question.component.js

```diff
@@ -0,0 +1,378 @@
+                vm.question.hint = questionTextParts[1].trim();
+                vm.question.explanation = questionTextParts[2].trim();
+            } else {
+                vm.question.hint = questionTextParts[2].trim();
+                vm.question.explanation = questionTextParts[1].trim();
+            }
+        } else if (questionText.indexOf("[-h]") !== -1) {
+            vm.question.hint = questionTextParts[1].trim();
+            vm.question.explanation = null;
+        } else if (questionText.indexOf("[-e]") !== -1) {
+            vm.question.hint = null;
+            vm.question.explanation = questionTextParts[1].trim();
+        } else {
+            vm.question.hint = null;
+            vm.question.explanation = null;
+        }
+    }
+
+    /**
+     * Parse the an answer markdown and apply the result to the question's data
+     *
+     * The markdown rules are as follows:
+     *
+     * 1. Text starts with [x] or [ ] (also accepts [X] and [])
+     *    => Answer options are marked as isCorrect depending on [ ] or [x]
+     * 2. The answer text is split at [-h] and [-e] tags.
+     *    => First part is text. Everything after [-h] is Hint, anything after [-e] is explanation
+     *
+     * @param text {string} the markdown text to parse
+     * @param answer {answerOption} the answer, where to save the result
+     */
+    function parseAnswerMarkdown(text, answer) {
+        text = text.trim();
+        // first split by [], [ ], [x] and [X]
+        var answerParts = text.split(/\[\]|\[ \]|\[x\]|\[X\]/g);
+        // work on answer options
+        // find the box (text in-between the parts)
+        var answerOption = {};
+        var answerOptionText = answerParts[1];
+        var startOfThisPart = text.indexOf(answerOptionText);
+        var box = text.substring(0, startOfThisPart);
+        // check if box says this answer option is correct or not
+        answer.isCorrect = (box === "[x]" || box === "[X]");
+
+        // parse this answerOption
+        answerParts = answerOptionText.split(/\[\-e\]|\[\-h\]/g);
+        answer.text = answerParts[0].trim();
+        if (answerOptionText.indexOf("[-h]") !== -1 && answerOptionText.indexOf("[-e]") !== -1) {
+            if (answerOptionText.indexOf("[-h]") < answerOptionText.indexOf("[-e]")) {
+                answer.hint = answerParts[1].trim();
+                answer.explanation = answerParts[2].trim();
+            } else {
+                answer.hint = answerParts[2].trim();
+                answer.explanation = answerParts[1].trim();
+            }
+        } else if (answerOptionText.indexOf("[-h]") !== -1) {
+            answer.hint = answerParts[1].trim();
+            answer.expalanation = null;
+        } else if (answerOptionText.indexOf("[-e]") !== -1) {
+            answer.hint = null;
+            answer.explanation = answerParts[1].trim();
+        } else {
+            answer.hint = null;
+            answer.explanation = null;
+        }
+    }
+
```

Comment:
The same applies here: CODE DUPLICATION IS BAD!
Please avoid it and restructure the code accordingly!